brian2tools/nmlimport/nml.py

```diff
@@ -0,0 +1,349 @@
+        """
+
+        sec = self._build_section(section, parent_section)
+        for s in section.sectionList:
+            sec[s.name] = self.build_morphology(s, section)
+        return sec
+
+    def get_segment_group_ids(self, group_id, morph):
+        """
+        Returns segment ids of all segments of a SegmentGroup with id
+        `group_id` present in .nml file.
+
+        Parameters
+        ----------
+        group_id: str
+            SegmentGroup's id/name whose information is required.
+        morph: Morphology
+            Brian's morphology object created from .nml file.
+        Returns
+        -------
+        list
+            List of segment ids
+        """
+
+        # Appends member segment ids to list.
+        def resolve_member(mem_list, members):
+            if members is not None:
+                for m in members:
+                    mem_list.append(m.segments)
+
+        # Resolves SegmentGroup's included inside parent SegmentGroup.
+        def resolve_includes(l, grp, m):
+            if grp.includes is not None:
+                for g in grp.includes:
+                    resolve_includes(l, self._get_segment_group(m,
+                                                                g.segment_groups),
+                                     m)
+            resolve_member(l, grp.members)
+
+        id_list = []
+        for g in morph.segment_groups:
+            if g.id == group_id:
+                resolve_includes(id_list, g, morph)
+                resolve_member(id_list, g.members)
+        return id_list
+
+    def get_resolved_group_ids(self, m):
+        """
+        Returns dictionary of relative ids(i.e ids used inside Brian's
+        morphology) of all segments in each SegmentGroup present in given
+        morphology object.
+
+        Parameters
+        ----------
+        m: Morphology
+            Morphology object whose resolved group ids we need.
+
+        Returns
+        -------
+        dict
+            A dictionary of resolved segment ids of each SegmentGroup,
+            here each SegmentGroup's id is a key of this dictionary.
+        """
+
+        # Returns id mappings of segments present in .nml file
+        def get_id_mappings(segments, parent_node=None, counter=0):
+            if parent_node is None:
+                parent_node = self._get_root_segment(segments).id
+
+            mapping = {}
+            children = get_child_segments(segments)
+            self._perform_dfs(mapping, parent_node, counter, children)
+            return mapping
+
+        resolved_ids = {}
+        for group in m.segment_groups:
+            grp_ids = self.get_segment_group_ids(group.id, m)
+            id_map = get_id_mappings(m.segments)
+            resolved_ids[group.id] = list(set([id_map[grp_id] for grp_id in
+                                               grp_ids]))
+        return resolved_ids
+
+    # Helper function to read .nml file and return document object
```

Comment:
Even though docstrings for helper functions are not mandatory, I'd put this comment as a docstring into the function (which makes it accessible to the help, etc.)